### DISCUSSION TITLE
Using an alternative dict for settings (not REST_FRAMEWORK)

### DISCUSSION THREAD

**Discussion by tobiasgardner** [encode/django-rest-framework#9351] [Question & Answer] [answered]
Hi! We already have an API in place based on plain DRF. But now we want to introduce a v2 of the API, based on the JSON:API module (https://github.com/django-json-api/django-rest-framework-json-api)

Both DRF and JSON:API module are using the dict REST_FRAMEWORK for settings. In order for our v1 API to continue working as is for the moment, we need to keep the existing settings in the REST_FRAMEWORK dictionary in settings.py and introduce a new dict for the JSON:API module.

Is it possible to rename our existing REST_FRAMEWORK dict to e.g. REST_FRAMEWORK_v1 and if so, how do we get DRF to use that?

Thanks in advance!

**Comment by lovelydinosaur** [comment, marked as answer]
We don't support that, no. Perhaps one approach would be running the different APIs as different processes with `runserver` pointed at a different Django settings module in each case? You'd also need to configure the frontend server (gunicorn or whatever) to route to each of the processes, doing that by hostname `v2.myapi.com` might be the reasonable approach.

(Just an idea, good luck with it)

**Reply by tobiasgardner** [reply]
Thanks for the answer and suggestion!
Would this not be a good feature? I guess that the need to develop a new version of the API which is not 100% backwards compatible will arise in many different projects, and enabling different dicts for the settings would be part of that.

**Reply by lovelydinosaur** [reply]
Not something we'd likely spend time on ourselves, no.
You're welcome to clone the repo and investigate adding this yourself.
